File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        st=[]
        dev=0
        for ch in s:
            if ch=="(":
                st.append(ch)
                dev=max(dev,len(st))
            elif ch==")":
                st.pop()
        return dev
        
        

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        
        int pass=0;
        int ans=0;
        for(char c :s){
            if(c=='('){
                pass++;
            }
            else if(c==')'){
                pass--;
                
            }
            ans=max(ans,pass);

        }
        return ans;
        
    }
};